fastlane/lanes/build.rb

```diff
@@ -24,7 +24,21 @@
       derived_data_path: DERIVED_DATA_PATH,
       build_for_testing: true,
       device: options[:device],
-      deployment_target_version: options[:ios_version]
+      deployment_target_version: options[:ios_version],
+      xcargs: {
+        # We treat warnings as errors in the codebase.
+        # The approach works well for builds from Xcode, but some React-pods fail when built from Fastlane.
+        #
+        # Adding this as a build setting in the `Podfile` `post_install` hook doesn't seem to work,
+        # so we're left to add it "globally" at this level.
+        #
+        # This approach is unfortunate because it leaves the door open for failures the were ignore locally to land in CI.
+        # It would be great if we could ge rid of it as some point, maybe once Gutenberg comes as an XCFramework?
+        #
+        # Interestingly, this workaround is required only when building for testing.
+        # That is, builds for App Store or AppCenter work fine.
```

Comment:
Actually... this made me think of something. I reverted the PR to draft while I try it out 🤔

Reply:
Yep. I forgot I never tried to run the tests locally. The issue was fixable at the project level, removing that awkward hack in the `Fastfile`